src/PKSim.Core/Services/HIDiseaseStateImplementation.cs

```diff
@@ -0,0 +1,220 @@
+      {
+         _valueOriginRepository = valueOriginRepository;
+         _formulaFactory = formulaFactory;
+         _ageDimension = dimensionRepository.AgeInYears;
+      }
+
+      public bool IsSatisfiedBy(DiseaseState diseaseState) => diseaseState.IsNamed(CoreConstants.DiseaseStates.HI);
+
+      public bool ApplyTo(Individual individual)
+      {
+         updateBloodFlows(individual, updateParameter);
+         updateVolumes(individual, updateParameter);
+         updateGFR(individual, updateParameter);
+
+         return true;
+      }
+
+      private void updateGFR(Individual individual, Action<IParameter, double, bool> updateParameterFunc)
+      {
+         var score = childPughScoreFor(individual);
+         var kidney = individual.Organism.Organ(KIDNEY);
+         var GFR_spec = kidney.Parameter(GFR_SPEC);
+         updateParameterFunc(GFR_spec, _gfrScalingFactor[score], true);
+      }
+
+      private void updateBloodFlows(Individual individual, Action<IParameter, double, bool> updateParameterFunc)
+      {
+         var score = childPughScoreFor(individual);
+         var organism = individual.Organism;
+         var updateBloodFlow = updateBloodFlowDef(updateParameterFunc, organism);
+
+         //PortalGF
+         var portalOrgans = new[] { STOMACH, SMALL_INTESTINE, LARGE_INTESTINE, SPLEEN, PANCREAS };
+         portalOrgans.Each(x => updateBloodFlow(x, _portalFlowScalingFactor[score]));
+
+         //Hepatic
+         updateBloodFlow(LIVER, _hepaticFlowScalingFactor[score]);
+
+         //Renal
+         updateBloodFlow(KIDNEY, _renalFlowScalingFactor[score]);
+
+         //other organs
+         var otherOrgans = new[] { BONE, FAT, GONADS, HEART, MUSCLE, SKIN };
+         otherOrgans.Each(x => updateBloodFlow(x, _otherOrgansFlowScalingFactor[score]));
+      }
+
+      private void updateVolumes(Individual individual, Action<IParameter, double, bool> updateParameterFunc)
+      {
+         var score = childPughScoreFor(individual);
+         var organism = individual.Organism;
+         var parameter = organism.Container(LIVER).Parameter(Constants.Parameters.VOLUME);
+         updateParameterFunc(parameter, _hepaticVolumeScalingFactor[score], true);
+      }
+
+      private Action<string, double> updateBloodFlowDef(Action<IParameter, double, bool> updateParameterFunc, IContainer organism) => (organName, factor) =>
+      {
+         var parameter = organism.Container(organName).Parameter(SPECIFIC_BLOOD_FLOW_RATE);
+         updateParameterFunc(parameter, factor, true);
+      };
+
+      private double childPughScoreFor(Individual individual)
+      {
+         return individual.OriginData.DiseaseStateParameters.FindByName(CHILD_PUGH_SCORE).Value;
+      }
+
+      public bool ApplyForPopulationTo(Individual individual)
+      {
+         throw new NotImplementedException();
+      }
+
+      public Individual CreateBaseIndividualForPopulation(Individual originalIndividual)
```

Comment:
@Yuri05 Almost done with Individual. NExt in population (hopefully similar to what we do for CKD)